doc/input.md

```diff
@@ -1,41 +1,84 @@
 # Input
 
-In order to handle user input, you can use the libraries provided by Flutter for regular apps: [Gesture Recognizers](https://flutter.io/gestures/).
-
-However, in order to bind them, use the `Flame.util.addGestureRecognizer` method; in doing so, you'll make sure they are properly unbound when the game widget is not being rendered, and so the rest of your screens will work appropriately.
-
-For example, to add a tap listener ("on click"):
+Inside `package:flame/gestures.dart` you can find a whole set of `mixin` which can be included on your game class instance to be able to receive touch input events. Bellow you can see the full list of these `mixin`s and its methods:
 
 ```dart
-    Flame.util.addGestureRecognizer(new TapGestureRecognizer()
-        ..onTapDown = (TapDownDetails evt) => game.handleInput(evt.globalPosition.dx, evt.globalPosition.dy));
+- TapDetector
+  - onTap
+  - onTapCancel
+  - onTapDown
+  - onTapUp
+
+- SecondaryTapDetector
+  - onSecondaryTapDown
+  - onSecondaryTapUp
+  - onSecondaryTapCancel
+
+- DoubleTapDetector
+  - onDoubleTap
+
+- LongPressDetector
+  - onLongPress
+  - onLongPressStart
+  - onLongPressMoveUpdate
+  - onLongPressUp
+  - onLongPressEnd
+
+- VerticalDragDetector
+  - onVerticalDragDown
+  - onVerticalDragStart
+  - onVerticalDragUpdate
+  - onVerticalDragEnd
+  - onVerticalDragCancel
+
+- HorizontalDragDetector
+  - onHorizontalDragDown
```

Comment:
one suggestion, see if it makes sense: to rename these so that they conflict. so for example this could be called onDragDown so it would conflict with the VerticalDragDetector and make using incompatible mixins a compile time error.

however due to how mixins work in dart I'm not so sure it would actually be a compiler error, so please take this suggestion with a grain of salt.

if this is not a good idea, current impl LGTM

Reply:
Yeah, I don't think it would give a compilation error, you have multiple mixins with the same name, and  the valid implementation will be last method from the last mixin.